### ISSUE TITLE
L.rectangle component ?

### ISSUE THREAD

**Issue by radcliff** [miguelcobain/ember-leaflet#12]
Any plans to add [`L.rectangle`](http://leafletjs.com/reference.html#rectangle)?

**Comment by miguelcobain** [comment]
Definitely.
Anything in core leaflet will be accepted in core ember-leaflet.

**Comment by miguelcobain** [comment]
## Add rectangle layer support to ember-leaflet.

API should be

```hbs
<layers.rectangle @bounds={{lat-lng-bounds (array 50 50) (array 51 51)}}/>
```

## 1. create `addon/components/rectangle-layer.js`

- `leafletRequiredOptions` is just `['bounds']`
- `leafletProperties` is also just `['bounds']`
- it should extend polygon-layer component and override the `createLayer` method to use `L.rectangle` instead

## 2. create the re-export file

- create the re-export file at `app/components/rectangle-layer.js`

## 3. make `<LeafletMap>` yield the new component

- on `addon/templates/current/leaflet-map.hbs` add the new rectangle-layer to the yield object using `rectangle=(component "rectangle-layer" parentComponent=this)`

## 4. add tests for rectangle layer

- tests should look similar to circle layer test (https://github.com/miguelcobain/ember-leaflet/blob/master/tests/integration/components/circle-layer-test.js)
- they should of course use the new layer and the [L.rectangle](https://leafletjs.com/reference-1.4.0.html#rectangle) methods

**Comment by miguelcobain** [comment]
Closed on #413 
4 years later 🎉
